### PR TITLE
Feat/add termination

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR adds the termination feature to the El Collectooorr.

### CODE REVIEW

File: packages/elcollectooorr/skills/elcollectooorr_abci/models.py [hunk 1]
```diff
@@ -142,8 +142,8 @@ def _get_decision_model_type(self, kwargs: dict) -> Type[ABC]:
         return valid_types[model_type]
 
 
-class Params(ElCollectooorParams, TransactionParams, FractionalizeDeploymentParams):
```
Comment (0xArdi) [inline packages/elcollectooorr/skills/elcollectooorr_abci/models.py:145]: The removal of `TransactionParams` is a side effect of having `TerminationParams` extend `TerminationParams`. 
I think this is more confusing like this, than having the `TransactionParams` be imported as a requirement for abci apps that don't have a direct dependency on `transaction_settlement_abci`. I think that most (not to say all) chained abci apps will have a direct dependency on `transaction_settlement_abci`.

File: packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py [hunk 2]
```diff
@@ -1629,7 +1633,9 @@ class ElCollectooorrFullRoundBehaviour(AbstractRoundBehaviour):
         *PostFractionsPayoutRoundBehaviour.behaviours,
         *TransferNFTAbciBehaviour.behaviours,
         *ResyncAbciBehaviour.behaviours,
+        *TerminationAbciBehaviours.behaviours,
     }
```
Comment (DavidMinarsch) [inline packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py:1637]: Why is this needed here? I thought the with_termination does this

File: packages/elcollectooorr/skills/elcollectooorr_abci/skill.yaml [hunk 3]
```diff
@@ -124,14 +125,14 @@ models:
...
-      on_chain_service_id: null
+      on_chain_service_id: 1
       period_setup: {}
       reset_tendermint_after: 60
       retry_attempts: 400
       retry_timeout: 3
       round_timeout_seconds: 120.0
       service_id: el_collectooorr_1
-      service_registry_address: null
+      service_registry_address: '0x48b6af7B12C71f09e2fC8aF4855De4Ff54e775cA'
```
Comment (DavidMinarsch) [inline packages/elcollectooorr/skills/elcollectooorr_abci/skill.yaml:135]: Why are you adding it?

File: packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py (hunk 2 above)
Comment (0xArdi) [reply packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py:1637]: `with_termination` is not implemented yet, also that would take care of the AbciApp part only. Although, I really don't like that extra effort is required on this side as well. I would suggest we make a `with_termination` decorator that is applied to the behaviour. The decorator would set both the AbciApp and Behaviour params.

File: packages/elcollectooorr/skills/elcollectooorr_abci/skill.yaml (hunk 3 above)
Comment (0xArdi) [reply packages/elcollectooorr/skills/elcollectooorr_abci/skill.yaml:135]: Its required by termination, and having the mainnet address as the default in this case I think is the right thing to do, because the El Collectoooorr is registered on mainnet.

File: packages/elcollectooorr/skills/elcollectooorr_abci/models.py (hunk 1 above)
Comment (Adamantios) [reply packages/elcollectooorr/skills/elcollectooorr_abci/models.py:145]: Yes, this seems confusing. What would you suggest? Adding the tx settlement dependency back?

**Review by Adamantios — APPROVED** [review]

File: packages/elcollectooorr/skills/elcollectooorr_abci/models.py (hunk 1 above)
Comment (0xArdi) [reply packages/elcollectooorr/skills/elcollectooorr_abci/models.py:145]: I think removing it from `TerminationParams` is the best option.

Comment (Adamantios) [reply packages/elcollectooorr/skills/elcollectooorr_abci/models.py:145]: I agree. 

What I meant is that it would require adding a dependency back as mentioned in https://github.com/valory-xyz/open-autonomy/pull/1459#discussion_r1011629337.

Comment (0xArdi) [reply packages/elcollectooorr/skills/elcollectooorr_abci/models.py:145]: Sorry I misunderstood you, yes that's what I was thinking as well.

File: packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py (hunk 2 above)
Comment (DavidMinarsch) [reply packages/elcollectooorr/skills/elcollectooorr_abci/behaviours.py:1637]: Please open an issue. I agree we should have a singular place where we add the background stuff.

File: packages/elcollectooorr/skills/elcollectooorr_abci/models.py (hunk 1 above)
Comment (DavidMinarsch) [reply packages/elcollectooorr/skills/elcollectooorr_abci/models.py:145]: I am not sure. It deserves further thought before we revert it. Let's track it in an issue first.

**Review by DavidMinarsch — APPROVED** [review]
